### PR TITLE
Add Ubuntu-18.04 to CI testing

### PR DESCRIPTION
**Pull request by baude**
Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

**Comment by baude** [comment]
If reviewers don't mind, let me iron things out before commenting...

File: libpod/stats.go [hunk 1]
```diff
@@ -33,13 +33,17 @@ func (c *Container) GetContainerStats(previousStats *ContainerStats) (*Container
 	if err != nil {
 		return nil, err
 	}
-
-	cgroup, err := cgroups.Load(cgroups.V1, cgroups.StaticPath(cgroupPath))
+	v1CGroups := GetV1CGroups(getExcludedCGroups())
+	if err != nil {
```
Comment (mheon) [inline libpod/stats.go]: No need for the error here, this doesn't return an error

File: libpod/stats.go [hunk 2]
```diff
@@ -33,13 +33,17 @@ func (c *Container) GetContainerStats(previousStats *ContainerStats) (*Container
 	if err != nil {
 		return nil, err
 	}
-
-	cgroup, err := cgroups.Load(cgroups.V1, cgroups.StaticPath(cgroupPath))
+	v1CGroups := GetV1CGroups(getExcludedCGroups())
```
Comment (mheon) [inline libpod/stats.go]: Need this change in runtime_pod_linux.go line 268

File: cmd/podman/utils.go [hunk 3]
```diff
@@ -215,3 +217,26 @@ func getPodsFromContext(c *cli.Context, r *libpod.Runtime) ([]*libpod.Pod, error
 	}
 	return pods, lastError
 }
+
+// GetV1CGroups gets the V1 cgroup subsystems and then "filters"
```
Comment (mheon) [inline cmd/podman/utils.go]: Is this duped from libpod/util.go?

**Comment by mheon** [comment]
Oops, sorry, didn't see your comment until now

File: libpod/stats.go (hunk 1 above)
Comment (baude) [reply libpod/stats.go]: fixed

File: test/e2e/libpod_suite_test.go [hunk 4]
```diff
@@ -148,7 +150,17 @@ func PodmanCreate(tempDir string) PodmanTest {
...
 	}
 
+	// Ubuntu doesn't use systemd cgroups
+	if host.Distribution == "ubuntu" {
+		cgroupManager = "cgroupfs"
+	}
+
 	runCBinary := "/usr/bin/runc"
+	altRunCBinary := "/usr/sbin/runc" // ubuntu
+	if _, err := os.Stat(runCBinary); err != nil {
```
Comment (giuseppe) [inline test/e2e/libpod_suite_test.go]: could we use `os.LookPath("runc")` here?

File: cmd/podman/utils.go (hunk 3 above)
Comment (baude) [reply cmd/podman/utils.go]: yeah too many util.go, fixed

File: test/e2e/libpod_suite_test.go (hunk 4 above)
Comment (baude) [reply test/e2e/libpod_suite_test.go]: given that /usr/sbin should be in the path, I guess it would work nicely ... everyone prefer that?

Comment (baude) [reply test/e2e/libpod_suite_test.go]: fixed

**Comment by jlebon** [comment]
The `makecache` failure is simply because Ubuntu is not supported in PAPR. For a list of supported platforms, see https://github.com/projectatomic/papr/blob/master/docs/DISTROS.md.

Thread about Ubuntu support: https://github.com/projectatomic/papr/issues/57. As mentioned there, trivial patches to get it working in Ubuntu would probably be accepted. Though at this point, I consider PAPR to be in maintenance mode until we decide the path forward (see https://github.com/projectatomic/papr/issues/62).

Suggestions:
- use `podman run --privileged docker.io/ubuntu:18.04 ...`
- if full virtualization is a hard requirement, you can try `curl`ing an Ubuntu cloud image and then running `qemu-kvm`. We can look into mirroring the image somewhere to make that more efficient.

**Comment by baude** [comment]
given that testing with papr is a no-go, i re-enabled travis and disabled the attempt to test in papr.  there is still relevant code here that fixes bonified problems on modern ubuntu distributions and will be critical to testing once we figure out how/where.

**Comment by baude** [comment]
bot, retest this please

**Comment by baude** [comment]
because we have two LGTMs, im going to /lgtm

**Pull request merged**